### PR TITLE
feat: unable to submit appeals with expired deadline date

### PR DESCRIPTION
**Pull request by dimodimovplanning**
stop users from submitting appeals with expired decision date

## Ticket Number
https://pins-ds.atlassian.net/browse/AS-2076

## Description of change
users will be unable to submit appeals with expired decision date

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/forms-web-app/src/controllers/appellant-submission/confirmation.js [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
+  const decisionDatePassed = isDecisionDatePassed(req.session.appeal);
 
-  res.render(VIEW.APPELLANT_SUBMISSION.CONFIRMATION, {
-    appellantEmail,
-    appealId,
-  });
+  if (decisionDatePassed) {
+    res.redirect(`/${VIEW.ELIGIBILITY.DECISION_DATE_PASSED}`);
+  } else {
+    req.session.appeal = null;
```
Comment (swdpcomputing) [inline packages/forms-web-app/src/controllers/appellant-submission/confirmation.js:12]: Is this simply to clean up? 

Also, I can't see a test for the path where it becomes nullified.

File: packages/forms-web-app/src/controllers/eligibility/decision-date.js [hunk 2]
```diff
@@ -8,6 +8,14 @@ const getDeadlineDate = (decisionDate) => {
   return addWeeks(endOfDay(decisionDate), 12);
 };
 
+const isDecisionDatePassed = (appeal) => {
```
Comment (swdpcomputing) [inline packages/forms-web-app/src/controllers/eligibility/decision-date.js:11]: I can't see a test for this function added....

Comment (a6software) [inline packages/forms-web-app/src/controllers/eligibility/decision-date.js:11]: please extract this, `require` back in, and add unit tests to cover

thanks

**Review by swdpcomputing — COMMENTED** [review]
Apologies if these comments / questions are irrelevant / obvious as I'm not up to date with what you're doing. I had some free time so thought I'd have a look and save someone else the time if possible.

**Comment by a6software** [comment]
Hi Dimo, 

I'm looking at https://pins-ds.atlassian.net/browse/AS-2002 and see AS-2077 is also in 'waiting for review' - please can you share that change also? From my point of view, that is the more important piece of this ticket as without it, we still leave open the hole of accepting invalid appeals directly to the API. This ticket addresses the forms web app / 'front end' stuff, but if e.g. the appeals questionaire modifies the appeal directly, we don't validate the decision date as per your new logic in that case.

Thanks

**Pull request closed without merging**